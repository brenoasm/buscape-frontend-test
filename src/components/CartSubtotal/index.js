import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { numberToReal } from "../../utils/converters";

const propTypes = {
  price: PropTypes.object
};

const defaultProps = {
  price: {
    installmentValue: 0,
    installments: 0,
    value: 0
  }
};

const StyledCartSubtotal = styled.div`
  display: flex;
  flex-direction: column;
  line-height: 1.4em;
`;

const CartSubtotal = ({ price }) => (
  <StyledCartSubtotal>
    <span>
      {price.installments}x R$ {numberToReal(price.installmentValue)}
    </span>
    <span>ou R$ {numberToReal(price.value)} Ã  vista</span>
  </StyledCartSubtotal>
);

CartSubtotal.propTypes = propTypes;
CartSubtotal.defaultProps = defaultProps;

export default CartSubtotal;
