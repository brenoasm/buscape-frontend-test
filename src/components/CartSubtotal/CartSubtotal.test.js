import React from "react";
import { shallow } from "enzyme";

import CartSubtotal from "./";

describe("CartSubtotal", () => {
  const price = {
    installmentValue: 389.9,
    installments: 10,
    value: 3509.1
  };

  it("matches the snapshot", () => {
    const wrapper = shallow(<CartSubtotal />);

    expect(wrapper).toMatchSnapshot();
  });

  it("matches the snapshot with price being passed", () => {
    const wrapper = shallow(<CartSubtotal price={price} />);

    expect(wrapper).toMatchSnapshot();
  });
});
