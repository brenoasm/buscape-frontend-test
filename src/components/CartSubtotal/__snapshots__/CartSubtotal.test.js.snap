// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartSubtotal matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartSubtotal
    price={
      Object {
        "installmentValue": 0,
        "installments": 0,
        "value": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span>
          0
          x R$ 
          0,00
        </span>,
        <span>
          ou R$ 
          0,00
           à vista
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            0,
            "x R$ ",
            "0,00",
          ],
        },
        "ref": null,
        "rendered": Array [
          0,
          "x R$ ",
          "0,00",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ou R$ ",
            "0,00",
            " à vista",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ou R$ ",
          "0,00",
          " à vista",
        ],
        "type": "span",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  line-height: 1.4em;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span>
            0
            x R$ 
            0,00
          </span>,
          <span>
            ou R$ 
            0,00
             à vista
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              0,
              "x R$ ",
              "0,00",
            ],
          },
          "ref": null,
          "rendered": Array [
            0,
            "x R$ ",
            "0,00",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ou R$ ",
              "0,00",
              " à vista",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ou R$ ",
            "0,00",
            " à vista",
          ],
          "type": "span",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  line-height: 1.4em;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CartSubtotal matches the snapshot with price being passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartSubtotal
    price={
      Object {
        "installmentValue": 389.9,
        "installments": 10,
        "value": 3509.1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span>
          10
          x R$ 
          389,90
        </span>,
        <span>
          ou R$ 
          3.509,10
           à vista
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            10,
            "x R$ ",
            "389,90",
          ],
        },
        "ref": null,
        "rendered": Array [
          10,
          "x R$ ",
          "389,90",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ou R$ ",
            "3.509,10",
            " à vista",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ou R$ ",
          "3.509,10",
          " à vista",
        ],
        "type": "span",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  line-height: 1.4em;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span>
            10
            x R$ 
            389,90
          </span>,
          <span>
            ou R$ 
            3.509,10
             à vista
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              10,
              "x R$ ",
              "389,90",
            ],
          },
          "ref": null,
          "rendered": Array [
            10,
            "x R$ ",
            "389,90",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ou R$ ",
              "3.509,10",
              " à vista",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ou R$ ",
            "3.509,10",
            " à vista",
          ],
          "type": "span",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  line-height: 1.4em;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
